backend/app/api/endpoints/knowledge.py

```
@@ -1509,3 +1599,715 @@ def _update_kb_summary_after_deletion(kb_id: int, user_id: int, user_name: str):
+                granted_at=p.granted_at,
+            )
+        )
+
+    return PermissionListResponse(total=len(items), items=items)
+
+
+@permission_router.post("/{kb_id}/permissions")
+async def add_knowledge_permissions(
+    kb_id: int,
+    data: PermissionCreate,
+    current_user: User = Depends(security.get_current_user),
```

Ruff (0.14.14)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)